app/main.py

```diff
@@ -1 +1,59 @@
-#  write your code here
+class Animal:
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True
+    ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        else:
+            return 0
+
+
+class Cat(Animal):
+    def __init__(
+            self,
+            name: str,
+            is_hungry: bool = True
+    ) -> None:
+        super().__init__(
+            name=name,
+            appetite=3,
+            is_hungry=is_hungry
+        )
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(
+            self,
+            name: str,
+            is_hungry: bool = True
+    ) -> None:
+        super().__init__(
+            name=name,
+            appetite=7,
+            is_hungry=is_hungry
+        )
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animal_list: list[Animal]) -> int:
+    return sum([animal.feed() for animal in animal_list])
```

Comment:
```suggestion
    return sum(animal.feed() for animal in animal_list)
```

Reply:
this variant is preferred for its improved memory efficiency, lazy evaluation, and cleaner syntax. It is a good practice to use generator expressions when the result is only needed for iteration or when applying functions like sum.

Reply:
I didn't think about that, thank you. It looks better